sqlite_utils/cli.py

```diff
@@ -1407,7 +1407,7 @@ def create_database(path, enable_wal, init_spatialite, load_extension):
 
     # load spatialite from expected locations and initialize metadata
     if init_spatialite:
-        db.init_spatialite()
+        db.init_spatialite(find_spatialite() or load_extension)
```

Comment:
I don't understand this pattern.

The `init_spatialite()` function already calls `find_spatialite()` if necessary:

https://github.com/simonw/sqlite-utils/blob/fc221f9b62ed8624b1d2098e564f525c84497969/sqlite_utils/db.py#L1161-L1165

The `load_extension` option is a bit different: It's actually a list of extensions passed using `--load-extension path` on the command-line, which is used by this code:

https://github.com/simonw/sqlite-utils/blob/fc221f9b62ed8624b1d2098e564f525c84497969/sqlite_utils/cli.py#L3037-L3047